lib/src/components/NxSearchDropdown/__tests__/NxSearchDropdown.test.tsx

```diff
@@ -4,225 +4,203 @@
+    expect(quickRender().getByRole('alert', { hidden: true })).toHaveAttribute('aria-hidden', 'true');
+    expect(quickRender({ searchText: 'foo', disabled: true }).getByRole('alert', { hidden: true }))
+        .toHaveAttribute('aria-hidden', 'true');
+    expect(quickRender({ searchText: 'foo', disabled: false }).getByRole('alert'))
+        .toHaveAttribute('aria-hidden', 'false');
+    expect(quickRender({ searchText: 'foo', disabled: undefined }).getByRole('alert'))
+        .toHaveAttribute('aria-hidden', 'false');
+    expect(quickRender({ searchText: 'foo', disabled: null }).getByRole('alert'))
+        .toHaveAttribute('aria-hidden', 'false');
+    expect(quickRender({ searchText: 'foo' }).getByRole('alert'))
+        .toHaveAttribute('aria-hidden', 'false');
   });
 
   it('sets aria-live on the dropdown menu to "polite"', function() {
-    expect(getShallow().find(NxDropdownMenu)).toHaveProp('aria-live', 'polite');
+    expect(quickRender().getByRole('alert', { hidden: true })).toHaveAttribute('aria-live', 'polite');
   });
 
   it('sets aria-busy on the dropdown menu if loading is true', function() {
-    expect(getShallow().find(NxDropdownMenu)).toHaveProp('aria-busy', false);
-    expect(getShallow({ loading: true }).find(NxDropdownMenu)).toHaveProp('aria-busy', true);
+    expect(quickRender().getByRole('alert', { hidden: true })).toHaveAttribute('aria-busy', 'false');
+    expect(quickRender({ loading: true }).getByRole('alert', { hidden: true })).toHaveAttribute('aria-busy', 'true');
   });
 
-  it('sets the nx-search-dropdown__menu class on the NxDropdownMenu', function() {
-    expect(getShallow({ searchText: 'foo' }).find(NxDropdownMenu)).toHaveClassName('nx-search-dropdown__menu');
-  });
+  it('sets the alert role on the dropdown menu when it is in loading, error, or empty states', function() {
+    expect(quickRender({ searchText: 'asdf', matches: [] }).getByRole('alert')).toBeInTheDocument();
+    expect(quickRender({ searchText: 'asdf', loading: true }).getByRole('alert')).toBeInTheDocument();
 
-  it('sets the menu role on the NxDropdownMenu when it contains results', function() {
-    expect(getShallow({ searchText: 'asdf', matches: [{ id: 'foo', displayName: 'bar' }] }).find(NxDropdownMenu))
-        .toHaveProp('role', 'menu');
-  });
+    const dropdownMenu = quickRender({ searchText: 'asdf', error: 'foo' }).getAllByRole('alert')[0];
+    expect(dropdownMenu).toBeInTheDocument();
 
-  it('sets the alert role on the NxDropdownMenu when it is in loading, error, or empty states', function() {
-    expect(getShallow({ searchText: 'asdf', matches: [] }).find(NxDropdownMenu)).toHaveProp('role', 'alert');
-    expect(getShallow({ searchText: 'asdf', loading: true }).find(NxDropdownMenu)).toHaveProp('role', 'alert');
-    expect(getShallow({ searchText: 'asdf', error: 'foo' }).find(NxDropdownMenu)).toHaveProp('role', 'alert');
+    const errorAlert = within(dropdownMenu).getByRole('alert');
```

Comment:
What are these two lines trying to test? Isn't `errorAlert` just going to end up being the same element as `dropdownMenu`?

Reply:
> What are these two lines trying to test? Isn't `errorAlert` just going to end up being the same element as `dropdownMenu`?

I am trying to assert that the `errorAlert (.nx-alert)` is going to be within the dropdown menu `(.nx-dropdown-menu)` that has a role of `alert` as well, based on this DOM structure:

![image](https://user-images.githubusercontent.com/7110862/215560399-687a1db3-5c96-4f13-831c-dcb0e8314973.png)

Reply:
> I am trying to assert that the errorAlert (.nx-alert) is going to be within the dropdown menu (.nx-dropdown-menu) that has a role of alert as well, based on this DOM structure:

Having multiple nested `alert` elements was not deliberate I don't think, and should be fixed.  Adjusting the line that computes `dropdownMenuRole` in `NxSearchDropdown` so that it does not add the `alert` role in the error state should do it.

Reply:
> > I am trying to assert that the errorAlert (.nx-alert) is going to be within the dropdown menu (.nx-dropdown-menu) that has a role of alert as well, based on this DOM structure:
> 
> Having multiple nested `alert` elements was not deliberate I don't think, and should be fixed. Adjusting the line that computes `dropdownMenuRole` in `NxSearchDropdown` so that it does not add the `alert` role in the error state should do it.

Changed here: 551d7a62b46f821f20b029b3babed2169fa559af

Reply:
> Changed here: https://github.com/sonatype/sonatype-react-shared-components/commit/551d7a62b46f821f20b029b3babed2169fa559af

I suspect `menu` isn't right for the error case.  I'd just not set a role on the `.nx-dropdown-menu` itself at all for that case

Reply:
> > Changed here: [551d7a6](https://github.com/sonatype/sonatype-react-shared-components/commit/551d7a62b46f821f20b029b3babed2169fa559af)
> 
> I suspect `menu` isn't right for the error case. I'd just not set a role on the `.nx-dropdown-menu` itself at all for that case

Changed and added test here: ae9b6c9dff97113a8bced5d2a1ba5ef1c34c72e2